src/devices/wifi/nm-iwd-manager.c

```diff
@@ -307,24 +332,105 @@ interface_added (GDBusObjectManager *object_manager, GDBusObject *object,
                  GDBusInterface *interface, gpointer user_data)
 {
 	NMIwdManager *self = user_data;
+	NMIwdManagerPrivate *priv = NM_IWD_MANAGER_GET_PRIVATE (self);
+	GDBusProxy *proxy;
+
+	if (!priv->running)
+		return;
 
-	set_device_dbus_object (self, interface, object);
+	g_return_if_fail (G_IS_DBUS_PROXY (interface));
+
+	proxy = G_DBUS_PROXY (interface);
+
+	if (!strcmp (g_dbus_proxy_get_interface_name (proxy),
+	             NM_IWD_DEVICE_INTERFACE)) {
+		set_device_dbus_object (self, proxy, object);
+		return;
+	}
+
+	if (!strcmp (g_dbus_proxy_get_interface_name (proxy),
+	             NM_IWD_KNOWN_NETWORK_INTERFACE)) {
+		KnownNetworkId *id;
+		KnownNetworkData *data;
+		NMIwdNetworkSecurity security;
+		gs_unref_variant GVariant *value;
+		const char *type_str;
+
+		value = g_dbus_proxy_get_cached_property (proxy, "Type");
+		if (!value)
+			return;
+
+		type_str = g_variant_get_string (value, NULL);
+		if (!strcmp (type_str, "open"))
+			security = NM_IWD_NETWORK_SECURITY_NONE;
+		else if (!strcmp (type_str, "psk"))
+			security = NM_IWD_NETWORK_SECURITY_PSK;
+		else if (!strcmp (type_str, "8021x"))
+			security = NM_IWD_NETWORK_SECURITY_8021X;
+		else
+			return;
+
+		g_variant_unref (value);
+		value = g_dbus_proxy_get_cached_property (proxy, "Name");
+		if (!value)
+			return;
+
+		id = known_network_id_new (g_variant_get_string (value, NULL), security);
```

Comment:
why is the `security` part of the ID? Shouldn't the ID be the D-Bus path of the known-network, instead of the `(name, security)` pair?

Reply:
Interesting. So how does IWD cope with multiple SSIDs being in the vicinity? Will it just pick one randomly/deterministically, and hide all others?

Reply:
It will have to assume they're multiple BSSes in one ESS / WLAN and treat them as one network.  The spec says SSID == ESSID, we don't currently try to detect conflicts and I'm not sure how much could actually be done there.

Note IWD only exposes whole networks to NM over dbus so we'll only select the best BSS when we're connecting.